tests/test__mongomock.py

```diff
@@ -1743,6 +1743,46 @@ def test__aggregate26(self):
         ]
         self.cmp.compare_ignore_order.aggregate(pipeline)
 
+    def test__aggregate27(self):
+        # group addToSet
+        pipeline = [
+            {"$group": {"_id": "$a", "nb": {"$addToSet": "$count"}}},
+            {"$sort": {"_id": 1}}
+        ]
+        # self.cmp.compare cannot be used as addToSet returns elements in an unpredictable order
+        aggregations = self.cmp.do.aggregate(pipeline)
+        expected = list(aggregations["real"])
+        result = list(aggregations["fake"])
+        assert len(result) == len(expected)
+        for i, elt in enumerate(result):
+            for k, v in elt.iteritems():
+                if type(v) is list:
+                    assert set(v) == set(result[i][k])
```

Comment:
It seems that you're comparing result ... and result! I think you meant to check expected. I suggest that you use `zip` instead of accessing by element:
```py
for expected_elt, result_elt in zip(expected, result):
  self.assertEqual(expected_elt.keys(), result_elt.keys())
  for key in result_elt:
    if isinstance(result_elt[key], list):
      self.assertCountEqual(result_elt[key], expected_elt[key], msg=key)
    else:
      self.assertEqual(result_elt[key], expected_elt[key], msg=key)
```